src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/InlineableStringsResourceNode.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+using ILCompiler.DependencyAnalysisFramework;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Represents a resource blob used by the SR class in the BCL.
+    /// If this node is present in the graph, it means we were not able to optimize its use away
+    /// and the blob has to be generated.
+    /// </summary>
+    internal sealed class InlineableStringsResourceNode : DependencyNodeCore<NodeFactory>
+    {
+        private readonly EcmaModule _module;
+
+        public const string ResourceAccessorTypeName = "SR";
+        public const string ResourceAccessorTypeNamespace = "System";
+        public const string ResourceAccessorGetStringMethodName = "GetResourceString";
+
+        public InlineableStringsResourceNode(EcmaModule module)
+        {
+            _module = module;
+        }
+
+        public override bool InterestingForDynamicDependencyAnalysis => false;
+
+        public override bool HasDynamicDependencies => false;
+
+        public override bool HasConditionalStaticDependencies => false;
+
+        public override bool StaticDependenciesAreComputed => true;
+
+        public static bool IsInlineableStringsResource(EcmaModule module, string resourceName)
+        {
+            if (!resourceName.EndsWith(".resources", StringComparison.Ordinal))
+                return false;
+
+            // TODO: we should grab this name from the SR class
```

Comment:
We should create an issue to track this - TODOs in code tend to die a lone death of abandonment.
I would even remove the TODO from the code and just replace with a description of what it's currently doing. And leave the future improvements to the issue.